website/src/layouts/base/header/HamburgerMenu.tsx

```diff
@@ -0,0 +1,213 @@
+            >
+                <button
+                    className='absolute right-3 top-4 z-50 cursor-pointer border-none bg-transparent'
+                    onClick={closeMenu}
+                >
+                    <Hamburger isOpen={isOpen} />
+                </button>
+                <div className='flex max-h-screen min-h-screen flex-col justify-between overflow-y-auto p-5'>
+                    <div>
+                        <header>
+                            <div className='h-10'>
+                                <a href='/'>Genspectrum</a>
+                            </div>
+                        </header>
+                        <div className='flex-grow divide-y-2 divide-solid divide-gray-300 border-b-2 border-t-2 border-solid border-gray-300'>
+                            <OffCanvasNavItem key='/' text='Pathogens' level='one' path='/' />
+                            {pathogenMegaMenuSections.map((section) => (
+                                <OffCanvasNavSection section={section} key={section.headline} />
+                            ))}
+                            <OffCanvasNavItem key='/data' text='Data Sources' level='one' path='/data' />
+                            <OffCanvasLogin showLoggedInState={showLoggedInState} session={session} />
+                        </div>
+                    </div>
+
+                    <div className='flex justify-end'>
+                        <a href='https://github.com/GenSpectrum'>
+                            <div className='iconify size-10 justify-end mdi--github'></div>
+                        </a>
+                    </div>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+export { HamburgerMenu };
+
+type IndentLevel = 'one' | 'two';
+
+type OffCanvasNavItemProps = {
+    text: string;
+    path: string | false;
+    level: IndentLevel;
+    type?: 'small';
+    open?: () => void;
+    id?: string;
+};
+
+function OffCanvasNavItem({ text, level, path, type, open, id }: OffCanvasNavItemProps) {
+    const height = type === 'small' ? 'py-1' : 'py-3';
+
+    const indent: { [K in IndentLevel]: string } = {
+        one: 'ml-4',
+        two: 'ml-8',
+    };
+
+    return (
+        <div>
+            <div className='flex items-center'>
+                <button
+                    onClick={open}
+                    id={id}
+                    className={`${indent[level]} ${height} ${level === 'one' ? 'font-bold' : ''}`}
+                >
+                    {path === false ? text : <a href={path}> {text}</a>}
+                </button>
+            </div>
+        </div>
+    );
+}
+
+type OffCanvasNavSectionProps = {
+    section: PathogenMegaMenuSection;
+};
+
+function OffCanvasNavSection({ section }: OffCanvasNavSectionProps) {
+    const [viewDiv, setViewDiv] = useState(false);
+    const reverseView = () => {
+        setViewDiv(!viewDiv);
+    };
```

Comment:
Instead of using JS for toggling this menu, we could a CSS-only solution, which is probably simpler. Daisy UI's collapse items are usually good for those things:
https://daisyui.com/components/collapse/